tests/serial/direct-mesh-access/DirectAccessWithDataInitialization.cpp

```diff
@@ -0,0 +1,185 @@
+#ifndef PRECICE_NO_MPI
+
+#include "testing/Testing.hpp"
+
+#include <precice/SolverInterface.hpp>
+#include <vector>
+
+// Test case for a direct mesh access while also using data initialization to test the integration of the two features.
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(Serial)
+BOOST_AUTO_TEST_SUITE(DirectMeshAccess)
+BOOST_AUTO_TEST_CASE(DirectAccessWithDataInitialization)
+{
+  PRECICE_TEST("SolverOne"_on(1_rank), "SolverTwo"_on(1_rank));
+
+  if (context.isNamed("SolverOne")) {
+    // Set up Solverinterface
+    precice::SolverInterface interface(context.name, context.config(), context.rank, context.size);
+    BOOST_TEST(interface.getDimensions() == 2);
+    constexpr int dim         = 2;
+    const int     ownMeshID   = interface.getMeshID("MeshOne");
+    const int     otherMeshID = interface.getMeshID("MeshTwo");
+    const int     readDataID  = interface.getDataID("Forces", ownMeshID);
+    const int     writeDataID = interface.getDataID("Velocities", otherMeshID);
+
+    std::vector<double> ownPositions = std::vector<double>({0.5, 0.25});
+    std::vector<int>    ownIDs(ownPositions.size() / dim, 0);
+    interface.setMeshVertices(ownMeshID, ownIDs.size(), ownPositions.data(), ownIDs.data());
+
+    std::array<double, dim * 2> boundingBox = std::array<double, dim * 2>{0.0, 1.0, 0.0, 1.0};
+    // Define region of interest, where we could obtain direct write access
+    interface.setMeshAccessRegion(otherMeshID, boundingBox.data());
+
+    std::vector<double> readData(ownIDs.size(), 0);
+    std::vector<double> writeData;
+
+    // writeData for initialization
+    // for (int i = 0; i < otherMeshSize; ++i) {  // @todo otherMeshSize not available yet. See https://github.com/precice/precice/issues/1583.
+    for (int i = 0; i < 1; ++i) {
+      writeData.emplace_back(2);
+    }
```

Comment:
Maybe make this a comment as well, as it doesn't have any meaning right now?

Reply:
I rearranged things a bit in efe5f3f4c8e465087f8c8c53a31454d1dbeaa6b9. We needed this call to actually increase the size of the vector. But I was not a big fan of doing it this way, so I changed it.